ablator/main/mp.py

```diff
@@ -147,32 +148,35 @@ def __init__(self, *args, run_config: ParallelConfig, **kwargs):
 
         self.run_config: ParallelConfig
         self.run_config = run_config
-        self.device = self.run_config.device
+        self.device=butils.parse_device(self.run_config.device)
         self.experiment_dir: Path = Path(run_config.experiment_dir)
         self.logger = FileLogger(path=self.experiment_dir / "mp.log")
         self.experiment_state: ExperimentState
         self.total_trials = self.run_config.total_trials
-
-        self.gpu_mem_bottleneck = min(get_gpu_max_mem())
-        if min(get_gpu_max_mem()) != max(get_gpu_max_mem()):
-            self.logger.warn(
-                f"Bottlenecked memory utilization by {self.gpu_mem_bottleneck}."
-            )
-        self.cpu: float = self._make_cpu()
-        self.gpu: float = self._make_gpu()
+        if(self.device.startswith("cuda")):
+            self.gpu_mem_bottleneck = min(get_gpu_max_mem())
+            if min(get_gpu_max_mem()) != max(get_gpu_max_mem()):
+                self.logger.warn(
+                    f"Bottlenecked memory utilization by {self.gpu_mem_bottleneck}."
+                )
+            self.gpu: float = self._make_gpu()
+            self.cpu: float = self._make_cpu()
```

Comment:
would it be better to refactor outside the if statement?

Reply:
fixed